src/mongo/tree/MongoDatabaseTreeItem.ts

```diff
@@ -20,24 +21,30 @@ export class MongoDatabaseTreeItem implements IAzureParentTreeItem {
 	public readonly connectionString: string;
 	public readonly databaseName: string;
 
-	public isConnected: boolean = false;
+	private _isConnected: boolean;
 
-	constructor(databaseName: string, connectionString: string) {
+	constructor(databaseName: string, connectionString: string, parentId: string) {
 		this.databaseName = databaseName;
 		this.connectionString = connectionString;
+		this._isConnected = ext.connectedMongoDB && ext.connectedMongoDB.id === `${parentId}/${databaseName}`;
```

Comment:
Seems like you could get rid of setting _isConnected here if you turned it into a property get. Then also wouldn't need refreshLabel.

Reply:
Done